docs/DevGuide/CreatingExecutables.md

```diff
@@ -0,0 +1,80 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Creating Executables {#dev_guide_creating_executables}
+
+There are several different types of executables that can be built:
+- An executable that uses Charm++ to run in parallel,
+  e.g. `Executables/ParallelInfo`
+- An executable that does not use Charm++ to run in parallel and supplies its
+  own `main`, e.g. `Executables/Benchmark`
+- An executable that uses Charm++ to run in parallel but supplies its own `main`
+- An executable that uses custom compilation or linking flags,
+  e.g. `DebugPreprocessor`
+- Executables used for evolutions or elliptic solves
+
+### Executable Using Charm++ for Parallelization
+
+### Executable Using Charm++ with Custom main()
+
+While this is technically possible, it has not been tested. We recommend using
+the Charm++ supplied main chare mechanism for the time being.
+
+### Executable Not Using Charm++
+
+An example of an executable that does not use Charm++ for parallelization but
+still can use all other infrastructure in SpECTRE is in
+`src/Executables/HelloWorldNoCharm`. The `CMakeLists.txt` file adds the
+executable and tells Charm++ not to add a `main()` by passing the link flags
+`-nomain-module -nomain`. This is done using CMake's `set_target_properties`:
+
+```
+set_target_properties(
+  ${EXECUTABLE}
+  PROPERTIES LINK_FLAGS "-nomain-module -nomain"
+  )
+```
+
+To add the executable as a target you must use CMake's `add_executable` function
+directly, not any of the SpECTRE-provided wrappers that add Charm++ parallelized
+executables. For example,
+
+```
+add_executable(
+  ${EXECUTABLE}
+  EXCLUDE_FROM_ALL
+  HelloWorld.cpp
+  )
+```
+
+You can link in any of the SpECTRE libraries by adding them to the
+`target_link_libraries`, for example:
+
+```
+target_link_libraries(
+  ${EXECUTABLE}
+  DataStructures
+  )
+```
+
+We recommend that you add a test that the executable properly runs by adding an
+input file to `tests/InputFiles` in an appropriate subdirectory.
+
+\warning
+Currently calling `Parallel::abort` results in a segfault deep inside Charm++
+code. However, the error messages from `ASSERT` and `ERROR` are still printed.
```

Comment:
This does not match what I see.  If I change the function in HelloWorld.cpp to
```c++
int main() {
  DataVector a{1.0, 2.3, 8.9};
  ASSERT(false, "Nope");
  Parallel::printf("%s\n", a);
}
```
I don't see any output except a "Segmentation fault" message from my shell.

Reply:
Can't make the same change in 2 PRs without conflicts. Once the RunSingleTest one is merged it'll show up.

Reply:
I rebased and tested that this works correctly.